sdk/python/feast/diff/registry_diff.py

```diff
@@ -109,23 +114,28 @@ def diff_registry_objects(
     assert current_proto.DESCRIPTOR.full_name == new_proto.DESCRIPTOR.full_name
     property_diffs = []
     transition: TransitionType = TransitionType.UNCHANGED
-    if current_proto.spec != new_proto.spec:
-        for _field in current_proto.spec.DESCRIPTOR.fields:
+
+    if isinstance(current_proto, DataSourceProto):
```

Comment:
Mind explaining what is happening in this if statement? What are the two different conditions that we support?

Reply:
it's a difference in how we're serializing the proto. Every other FCO has a "spec" vs "meta" middle layer, but data source doesn't have that and has everything flattened.